### DISCUSSION TITLE
sqlite3 VACUUM when shutting down container

### DISCUSSION THREAD

**Discussion by brtwrst** [dani-garcia/vaultwarden#1729] [Q&A] [answered]
Greetings,

I'm using vaultwarden/server:alpine on my raspberry pi and I have an automated backup.
The backup consists of `shutting down the container` ->  `mounting the docker volume that has all the vaultwarden data into a new container` -> `taring the contents and saving the tar on a remote server`.

When (gracefully) shutting down the container the sqlite3 database is not "VACUUMed" and the `sqlite3-shm` and `sqlite3-wal` files are still present in the folder. These files increase the size of the backup.

Is there a way to automatically run the sqlite3 VACUUM when the container is shut down?

**Comment by BlackDex** [comment, marked as answer]
Not from the application side since with the current rocket implementation we can't detect or gracefully shutdown.

Also, its probably better to use `VACUUM INTO` as is done with the backup button within the admin. That will generate a new sqlite file which will be vacuumed. You don't even need to shutdown the container, just backup everything of the data folder while running.

**Comment by brtwrst** [comment]
Ok thank you.
